src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSASignatureDeformatter.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Security.Cryptography
+{
+    public class DSASignatureDeformatter : AsymmetricSignatureDeformatter
+    {
+        private DSA _dsaKey;
+
+        public DSASignatureDeformatter() { }
+
+        public DSASignatureDeformatter(AsymmetricAlgorithm key) : this()
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            _dsaKey = (DSA)key;
+        }
+
+        public override void SetKey(AsymmetricAlgorithm key)
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            _dsaKey = (DSA)key;
+        }
+
+        public override void SetHashAlgorithm(string strName)
+        {
+            try
+            {
+                // Verify the name is a valid HashAlgorithm even though it is not currently used
+                Oid.FromFriendlyName(strName, OidGroup.HashAlgorithm);
```

Comment:
Please make this just check that the input is a legal identifier for SHA1 to match Desktop.

Desktop currently only supports SHA1, and we should be thoughtful when that behavior is changed.  Do we allow only FIPS 186-3-acceptable algorithms (SHA-1 and SHA-2-x)? Do we widen it up to all resolvable HashAlgorithms?  Do we just get rid of the check altogether?

We should keep porting differences to a minimum; particularly when it's not due to not being able to easily replicate a behavior on Unix.